src/Lykke.Job.CandlesHistoryWriter.Services/Candles/CandlesCacheInitalizationService.cs

```diff
@@ -45,10 +45,14 @@ public class CandlesCacheInitalizationService : ICandlesCacheInitalizationServic
             // Depending on cache invalidation period and on asset pairs amount, there may be a case when
             // the invalidation timer fires before the cache loading has stopped. This will be a signal 
             // to skip timer-based invalidation.
-            if (InitializationState == CacheInitializationState.InProgress)
-                return;
+            var initLock = new object();
+            lock (initLock)
```

Comment:
This lock do nothing but just spends CPU cycles. Having code like this, each thread will lock its own initLock, since initLock is a local variable. initLock should be field of the CandlesCacheInitalizationService to preofrm cross-threads lock.